guard/src/rules/eval.rs

```diff
@@ -407,6 +407,11 @@ struct UnResolvedRhs<'r> {
     lhs: &'r PathAwareValue,
 }
 
+pub(crate) struct FileData {
+    pub(crate) path_aware_value: Result<PathAwareValue>,
```

Comment:
A bit of a nit... and i hate to be that guy, but since we already have a struct named DataFile, I think it might be nice to use a different name for this struct to help avoid any confusion in the future...Maybe something like NamedMergedData ? 

I also think its not the end of the world if we keep it as is, but I am 100% gonna mix the 2 up at some point in the future lol